android/BOINC/app/src/main/java/edu/berkeley/boinc/EventLogActivity.java

```diff
@@ -44,29 +45,29 @@
 import android.widget.ListView;
 import android.widget.Toast;
 
+import org.eclipse.collections.api.list.MutableList;
+import org.eclipse.collections.impl.list.mutable.FastList;
+
 public class EventLogActivity extends AppCompatActivity {
 
     private IMonitor monitor;
     private Boolean mIsBound = false;
 
-    public EventLogClientFragment clientFrag;
     public ListView clientLogList;
     public ClientLogListAdapter clientLogListAdapter;
-    public ArrayList<Message> clientLogData = new ArrayList<>();
+    public List<Message> clientLogData = new ArrayList<>();
 
-    public EventLogGuiFragment guiFrag;
     public ListView guiLogList;
     public ArrayAdapter<String> guiLogListAdapter;
-    public ArrayList<String> guiLogData = new ArrayList<>();
+    public List<String> guiLogData = new ArrayList<>();
 
-    private ArrayList<EventLogActivityTabListener<?>> listener = new ArrayList<>();
+    private MutableList<EventLogActivityTabListener<?>> listeners = new FastList<>();
```

Comment:
I believe here should be a generic List type, isn't it?

Reply:
MutableList is an interface that extends the List interface and is part of EC.

Reply:
https://www.eclipse.org/collections/javadoc/10.2.0/org/eclipse/collections/api/list/MutableList.html